src/cudadev/CondFormats/SiPixelROCsStatusAndMappingWrapper.cc

```diff
@@ -13,43 +13,40 @@
 #include "CUDACore/host_unique_ptr.h"
 #include "CUDADataFormats/gpuClusteringConstants.h"
 #include "CondFormats/SiPixelROCsStatusAndMappingWrapper.h"
+#include "CUDACore/copyAsync.h"
 
 SiPixelROCsStatusAndMappingWrapper::SiPixelROCsStatusAndMappingWrapper(SiPixelROCsStatusAndMapping const& cablingMap,
                                                                std::vector<unsigned char> modToUnp)
     : modToUnpDefault(modToUnp.size()), hasQuality_(true) {
-  cudaCheck(cudaMallocHost(&cablingMapHost, sizeof(SiPixelROCsStatusAndMapping)));
-  std::memcpy(cablingMapHost, &cablingMap, sizeof(SiPixelROCsStatusAndMapping));
-
+  // TODO: check if cudaStreamDefault is appropriate
+  auto cablingMapMetadata = cablingMap.soaMetadata();
+  cablingMapHostBuffer = cms::cuda::make_host_unique<std::byte[]>(cablingMapMetadata.byteSize(), cudaStreamDefault);
```

Comment:
Not really in principle, but it probably won't matter in practice. I'm going to rework the EventSetup side in this fall to allow the use of these smart pointers more naturally here.

Reply:
OK, so leaving is as-is, thanks.